src/sage/topology/moment_angle_complex.py

```diff
@@ -814,3 +819,712 @@ def has_trivial_lowest_deg_massey_product(self):
+        elements.remove(self.one())
+        # We create a dictionary, in which the keys are
+        # base elements, and values are lists, which
+        # represent the elements whose product with
+        # the key gives us zero
+        memo = {}
+        for base_element in elements:
+            memo[base_element] = set()
+            for x in elements:
+                if base_element * x == self.zero():
+                    memo[base_element].add(x)
+
+        max_length = 1
+        # The multiplication is commutative up to a sign,
+        # so we will not worry about the order in which
+        # we multiply the elements
+        for base_element in elements:
+            prod = base_element
+            length = 1
+            # We will avoid the elements which give us zero
+            avoid = memo[base_element]
+            for x in elements:
+                if x not in avoid and prod * x != self.zero():
+                    prod = prod * x
+                    length = length + 1
+                    # We further restrict the possible elements
+                    avoid = avoid.union(memo[x])
+
+            if length > max_length:
+                max_length = length
+
+        return max_length
+
+    class Element(CombinatorialFreeModule.Element):
+        def to_cycle(self):
+            r"""
+            Return the cocycle representative of ``self``.
+
+            The cohomology class gets lifted to its cococyle
+            representative in the cochain complex of the
+            corresponding simplicial complex given in the main
+            isomorphism.
+
+            .. NOTE::
+
+                The cocycle is practically useless without
+                knowing from which subcomplex it origintes.
+                We can use the :meth:`get_simplicial_complex`
+                method to retrieve that information.
+
+            EXAMPLES::
+
+                sage: Z = MomentAngleComplex([[0,1], [1,2], [2,0],
+                ....:                         [1,3], [3,4], [4,1]])
+                sage: H = Z.cohomology_ring()
+                sage: a = H.basis()[5, 1]
+                sage: a.to_cycle()
+                \chi_(3, 4)
+                sage: b = H.basis()[6, 3]
+                sage: b.to_cycle()
+                \chi_(3, 4)
+                sage: H.one().to_cycle()
+                \chi_()
+
+            We can obtain the originating simplicial complex
+            by using :meth:`get_simplicial_complex`::
+
+                sage: a.get_simplicial_complex()
+                Simplicial complex with vertex set (1, 3, 4) and
+                facets {(1, 3), (1, 4), (3, 4)}
+                sage: H.one().get_simplicial_complex()
+                Simplicial complex with vertex set () and facets {()}
+            """
+            if not self.is_homogeneous():
+                raise ValueError("only defined for homogeneous elements")
+            return sum(c * self.parent()._to_cycle_on_basis(i) for i, c in self)
```

Comment:
```suggestion
            P = self.parent()
            ret_par = ???
            return ret_par.linear_combination((P._to_cycle_on_basis(i), c) for i, c in self)
```
Perhaps this is not possible, but it would be good for the output to not depend on the element itself.